### ISSUE TITLE
Accessibility issues with navigation menu

### ISSUE THREAD

**Issue by roelvangils** [digipolisantwerp/acpaas-ui_angular#230]
Erik and Roel reported back these issues.

### 1.1.1 Non-text Content

- Issue: the arrows that are used to indicate a closed or opened sub-menu do not have a state or accessible name and fall for many of the [fontawesome](https://www.notion.so/Digipolis-5ed9b0aec4ed4f0e8d3684bdaa16c447#aea9265ed3b34ae3b24ec3e542235dc8) issues.

### 1.3.1 Info and Relationships

- Issue: the menu looks like a list of links but isn't presented as such
Fix: make the menu an `<ul>` with `<li>`'s and `<a>`'s. For example:

<ul>
<li>
<a href="/menu-target>Menu Target</a>
</li>
</ul>
- Issue: the fold-out link looks like a link but it performs an action.
Fix: Use a `<button>` instead of an `<a>`.

### 1.3.2 Meaningful Sequence

- Issue: the "mobile mode" of the menu appears in the code before the content, while visually it appears after the content. This can confuse and disorient users that navigate with keyboard.
Fix: Move the mobile version of the menu after the content or create a menu that can be put before content (so the skip link stays functional).

### 1.4.11 Non-text Contrast

- Issue: the hover state is hard to distinguish from the background
Fix: make the hover state easier to see. For example, make the text bold, add an icon or change the colors.

### 4.1.2 Name, Role, Value

- Issue: states like expanded sub-menus are not communicated
Fix: use `<button>`s with `aria-expanded` on them.

## Recommendations

The mobile version does not work for me. Most recent Firefox, Edge and Chrome in Windows 10. Pressing "More..." does nothing. Unable to test properly.

There is no demo of the "linking sub-menu" that does have a screenshot.

The mobile version has an animation that might disorient users and create vertigo. Please disable or [use the reduce-motion query in CSS](https://www.w3.org/WAI/WCAG21/Techniques/css/C39.html) to prevent it for certain users.

The mobile menu acts like a modal but does not follow the modal pattern

The menu can technically be used with a screen reader but the user hears no difference between a normal menu item, or an item that will fold out a sub-menu.

The `<nav>` does not need to be inside the `<main>` as it is navigation for more than just the `<main>`.

When navigating, the focus stays on the menu(item). This is unlike normal navigation between webpages and an unusual pattern.

The menu does not make clear what the current page is. Suggest adding this both visually and with the attribute `aria-current`.

You're not using `role="menu"`. This is consistent with other places on the web. Please continue to **not** use this role.

"More..."-menu sometimes opens when making window smaller.

**Comment by MichaelCastiau** [comment]
@TriangleJuice Almost all issues presented here cannot be fixed by simple touchups but would entice a *complete rewrite* of the component, introducing breaking changes.
